main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.Toolbox/MacToolbox.cs

```diff
@@ -38,13 +38,63 @@
 using CoreGraphics;
 using MonoDevelop.Components;
 using MonoDevelop.Components.Mac;
+using Foundation;
 
 namespace MonoDevelop.DesignerSupport.Toolbox
 {
-	class MacToolbox : NSStackView, IPropertyPadProvider, IToolboxConfiguration
+	class KeyViewLoopDelegate
 	{
-		const string ToolboxItemContextMenuCommand = "/MonoDevelop/DesignerSupport/ToolboxItemContextMenu";
+		readonly List<NSView> viewsKeyLoopOrder = new List<NSView> ();
+
+		public bool Contains (NSView view) => viewsKeyLoopOrder.Contains (view);
+
+		NSView GetNextFocusableViewForView (NSView view, int nextPositionInArray = 1)
+		{
+			if (nextPositionInArray == 0)
+				return view;
+
+			var currentIndex = viewsKeyLoopOrder.IndexOf (view);
+			if (currentIndex == -1) {
+				return null;
+			}
+
+			//nextPositionInArray is the next view we want to loop
+			var initialPosition = currentIndex + nextPositionInArray;
+
+			//this next view could be hidden then we want to iterate secuentially to next in this case until end of our array
+			var increment = nextPositionInArray >= 0 ? +1 : -1;
+
+			//with the initial position and increment we return the first view which can be first responder and it's not hidden
+			for (int j = initialPosition; j > 0 && j <= viewsKeyLoopOrder.Count - 1; j += increment) {
```

Comment:
This is still not going to check the first element:

```suggestion
			for (int j = initialPosition; j >= 0 && j <= viewsKeyLoopOrder.Count - 1; j += increment) {
```

Reply:
fixed !